executor/src/witgen/multiplicity_column_generator.rs

```diff
@@ -104,28 +112,55 @@ impl<'a, T: FieldElement> MultiplicityColumnGenerator<'a, T> {
             })
             .collect::<BTreeMap<_, _>>();
 
+        // A map from multiplicity column ID to the vector of multiplicities.
+        let mut multiplicity_columns = receive_infos
+            .values()
+            .map(|info| (info.multiplicity_column, vec![0; info.size]))
+            .collect::<BTreeMap<_, _>>();
+
         // Increment multiplicities for all bus sends.
-        for (bus_send, bus_receive) in identities.iter().filter_map(|i| match i {
-            Identity::BusSend(bus_send) => receive_infos
-                .get(&bus_send.bus_id().unwrap())
-                .map(|bus_receive| (bus_send, bus_receive)),
+        let bus_sends = identities.iter().filter_map(|i| match i {
+            Identity::BusSend(bus_send) => match bus_send.bus_id() {
+                // As a performance optimization, already filter out sends with a static
+                // bus ID for which we know we don't need to track multiplicities.
```

Comment:
On Keccak-via-RISC-V, this reduces multiplicity witgen from ~520ms to ~450ms (2.6% of witgen time). Before this PR, it's ~320ms. I suppose looking up the receive for each row below causes this regression. We could do more shenanigans to speed this up again in the static case, but I don't think it's worth it.